### PR TITLE
function to get default route interface

### PR DESCRIPTION
**Pull request by smruti77**
add get_default_route_interface() in host.py to get default
interface name.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Review by beraldoleal — CHANGES_REQUESTED** [review]
This is returning the first dev of the output list. Like I said on my previous comment if we have multiple defaults gateways this should return a list, not a single dev.

**Comment by pevogam** [comment]
Has anyone thought about simply migrating the networking utilities from Avocado VT or methods/parts of them? It is quite possible that there is already a function like this there but I am just wondering.

**Comment by beraldoleal** [comment]
> Has anyone thought about simply migrating the networking utilities from Avocado VT or methods/parts of them? It is quite possible that there is already a function like this there but I am just wondering.

That is nice to check, yes. But this `network/` module was completely redesigned a few months ago and its usage is a little simplified based only on two classes, so far: Host() and Interface().

**Comment by pevogam** [comment]
> > Has anyone thought about simply migrating the networking utilities from Avocado VT or methods/parts of them? It is quite possible that there is already a function like this there but I am just wondering.
> 
> That is nice to check, yes. But this `network/` module was completely redesigned a few months ago and its usage is a little simplified based only on two classes, so far: Host() and Interface().

The current simplified implementation is another reason it would be good to migrate complete code from there instead of tens of PRs to follow that will gradually end up at the same target destination. But these are just ideas.

File: avocado/utils/network/hosts.py [hunk 1]
```diff
@@ -64,6 +65,19 @@ def get_interface_by_ipaddr(self, ipaddr):
             if ipaddr in interface.get_ipaddrs():
                 return interface
 
+    def get_default_route_interface(self):
+        """Get the default route interface
+
+        :return: interface name
```
Comment (beraldoleal) [inline avocado/utils/network/hosts.py:71]: Return a list of ifnae names.

File: avocado/utils/network/hosts.py [hunk 2]
```diff
@@ -64,6 +65,19 @@ def get_interface_by_ipaddr(self, ipaddr):
             if ipaddr in interface.get_ipaddrs():
                 return interface
 
+    def get_default_route_interface(self):
+        """Get the default route interface
```
Comment (beraldoleal) [inline avocado/utils/network/hosts.py:69]: Get a list of default routes interfaces.

**Review by beraldoleal — CHANGES_REQUESTED** [review]

**Comment by beraldoleal** [comment]
> > > Has anyone thought about simply migrating the networking utilities from Avocado VT or methods/parts of them? It is quite possible that there is already a function like this there but I am just wondering.
> > 
> > 
> > That is nice to check, yes. But this `network/` module was completely redesigned a few months ago and its usage is a little simplified based only on two classes, so far: Host() and Interface().
> 
> The current simplified implementation is another reason it would be good to migrate complete code from there instead of tens of PRs to follow that will gradually end up at the same target destination. But these are just ideas.

Yes, definitely I will check the methods implemented there.

**Comment by clebergnu** [comment]
> Has anyone thought about simply migrating the networking utilities from Avocado VT or methods/parts of them? It is quite possible that there is already a function like this there but I am just wondering.

Hi @pevogam, this is the path that I suggested on this mailling list post:

https://www.redhat.com/archives/avocado-devel/2020-September/msg00006.html

It'd be nice to get feedback and some general consensus so we can start working towards it.

**Comment by pevogam** [comment]
> > Has anyone thought about simply migrating the networking utilities from Avocado VT or methods/parts of them? It is quite possible that there is already a function like this there but I am just wondering.
> 
> Hi @pevogam, this is the path that I suggested on this mailling list post:
> 
> https://www.redhat.com/archives/avocado-devel/2020-September/msg00006.html
> 
> It'd be nice to get feedback and some general consensus so we can start working towards it.

Hi @clebergnu, I would love to respond as soon as I can but I am still catching up with the 82.0 release of our plugin (it follows your release schedule) and I am a bit behind there due to more time spent upstream and pushing through with additional PRs. If you have more time I would be glad to see some of your reactions on some of my individual troubles and blockades where I have pinged you explicitly.

I commented faster here since recently I saw a lot of tendency in developing duplicate functionality when it comes to networking that is already available in VT. I might be wrong though, I have only touched the `utils_net.py` on the surface.

**Review by beraldoleal — APPROVED** [review]
Thanks for the changes @smruti77 , this LGTM.

IMO, this is an improvement for what we have today in `utils_net.py` on avocado-vt side, since there is not returning a proper list in case multiple default routes.

**Comment by beraldoleal** [comment]
@pevogam I share your concern and thanks for your quick input.

I will take care of migrating `utils_net.py` for the new place as suggested by @clebergnu on that thread. But I would like to follow a more OOP approach, like this one here.

**Comment by beraldoleal** [comment]
@smruti77 just one small error with isort. Travis is complaining:

 (003/952) selftests/isort.sh:  FAIL (4.02 s)

**Comment by pevogam** [comment]
Hi @beraldoleal,

> @pevogam I share your concern and thanks for your quick input.
> 
> I will take care of migrating `utils_net.py` for the new place as suggested by @clebergnu on that thread. But I would like to follow a more OOP approach, like this one here.

Following a more OOP approach or additional improvements don't contradict the argument of reuse. You could convert to OOP while migrating the code from there for instance. One way to confirm the development done here is superior to the development done there for instance would be to gradually deprecate their methods for the sake of the ones here and thus making sure by observation that all functional requirements on their side is already covered by avocado itself.

**Comment by beraldoleal** [comment]
> Hi @beraldoleal,
> 
> > @pevogam I share your concern and thanks for your quick input.
> > I will take care of migrating `utils_net.py` for the new place as suggested by @clebergnu on that thread. But I would like to follow a more OOP approach, like this one here.
> 
> Following a more OOP approach or additional improvements don't contradict the argument of reuse. You could convert to OOP while migrating the code from there for instance. One way to confirm the development done here is superior to the development done there for instance would be to gradually deprecate their methods for the sake of the ones here and thus making sure by observation that all functional requirements on their side is already covered by avocado itself.

I think that I didn't express myself in the proper way. I agree with you, and we are talking about the same thing ;)


> In any case, I was just adding a minor note that could be of use to you and the rest of the avocado maintainers. I don't mean to spark a bigger conversation, merely to hint at more beneficial possibilities.

No worries. It was a valid point/reminder.

**Comment by smruti77** [comment]
> @smruti77 just one small error with isort. Travis is complaining:
> 
> (003/952) selftests/isort.sh: FAIL (4.02 s)
done

**Pull request merged**